### PR TITLE
add my engineering QoL helper scripts

### PR DESCRIPTION
**Pull request by 0xdevalias**
- `scripts/deploy-master-to-prod-envs`: Basic hacky bash script to deploy `master` to all(?) of our environments
- `scripts/pr-show-commits`: Uses the GitHub CLI to call the GitHub API to retrieve all of the commits for a given PR (useful for quickly making the list of included commits for the description in deployment PRs)
- `scripts/pr-unresolved-review-comments`: Uses the GitHub CLI to call the GitHub GraphQL API to list the unresolved review comments for the specified PR

### CODE REVIEW

File: scripts/pr-show-commits [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# https://cli.github.com/
```
Comment (0xdevalias) [inline scripts/pr-show-commits:1]:
```suggestion
#!/usr/bin/env sh

# https://cli.github.com/
```

File: scripts/pr-unresolved-review-comments [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# https://cli.github.com/
```
Comment (0xdevalias) [inline scripts/pr-unresolved-review-comments:1]:
```suggestion
#!/usr/bin/env sh

# https://cli.github.com/
```

File: scripts/pr-show-commits [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+# https://github.com/octokit/core.js
+#   TODO: rewrite this so that it uses octokit core, and then maybe do it as a GitHub action/similar?
+#   https://github.com/octokit/graphql.js
+#   https://github.com/octokit/action.js
+#   https://github.com/octokit/request-action
+#   https://github.com/actions/toolkit
```
Comment (0xdevalias) [inline scripts/pr-show-commits:7]: [note] This is more future TODO musings, and not blocking for this PR

File: scripts/pr-unresolved-review-comments [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+# https://github.com/octokit/core.js#graphql-example
+#   TODO: rewrite this so that it uses octokit core, and then maybe do it as a GitHub action/similar?
+#   https://github.com/octokit/graphql.js
+#   https://github.com/octokit/action.js
+#   https://github.com/octokit/request-action
+#   https://github.com/actions/toolkit
+
+# Posted solution to:
+#   https://stackoverflow.com/questions/55713929/list-all-unresolved-pull-request-comments/66072198#66072198
+#   https://github.community/t/state-of-conversation-marked-resolved/14355/3?u=0xdevalias
```
Comment (0xdevalias) [inline scripts/pr-unresolved-review-comments:13]: [note] This is more future TODO musings, and not blocking for this PR

**Review by 0xdevalias — COMMENTED** [review]
[note] PR notes

File: scripts/pr-show-commits (hunk 1 above)
Comment (0xdevalias) [reply scripts/pr-show-commits:3]: fixed in https://github.com/sparkletown/sparkle/pull/1466/commits/e3545f0c0930d4401583e973b2c8b26f7639b439

File: scripts/pr-unresolved-review-comments (hunk 2 above)
Comment (0xdevalias) [reply scripts/pr-unresolved-review-comments:3]: fixed in https://github.com/sparkletown/sparkle/pull/1466/commits/e3545f0c0930d4401583e973b2c8b26f7639b439

**Review by yurii-lubynets — APPROVED** [review]
LGTM, It's a massive improvement for the deployment process

**Pull request merged**